### PR TITLE
Prevent Errors when transition is called after Component is reset

### PR DESCRIPTION
**Pull request by sscholle**
Sometimes a transition may be triggered when the state is RESET (null) - add check for this

### CODE REVIEW

**Comment by sscholle** [comment]
@robguy21 you're up!

File: src/machine.js [hunk 1]
```diff
@@ -44,8 +44,16 @@ class Machine {
   }
 
   getEdgeFromTransition(transition: TransitionObject): Edge {
+    if (!this.#state) {
+      this.log('Invalid State - unable get Edges for Transistion');
+      return false;
+    }
+    if (!transition) {
+      this.log('Invalid Transition parameter passed');
+      return false;
```
Comment (robguy21) [inline src/machine.js]: this breaks the expected return- you should throw an error

File: src/machine.js [hunk 2]
```diff
@@ -44,8 +44,16 @@ class Machine {
   }
 
   getEdgeFromTransition(transition: TransitionObject): Edge {
+    if (!this.#state) {
+      this.log('Invalid State - unable get Edges for Transistion');
+      return false;
+    }
+    if (!transition) {
```
Comment (robguy21) [inline src/machine.js]: this should be handled in `verifyEdge`, you should rather check that `transition` supports a `.toString()` call

File: src/machine.js [hunk 3]
```diff
@@ -44,8 +44,16 @@ class Machine {
   }
 
   getEdgeFromTransition(transition: TransitionObject): Edge {
+    if (!this.#state) {
+      this.log('Invalid State - unable get Edges for Transistion');
+      return false;
```
Comment (robguy21) [inline src/machine.js]: rather throw

File: src/machine.js [hunk 4]
```diff
@@ -44,8 +44,16 @@ class Machine {
   }
 
   getEdgeFromTransition(transition: TransitionObject): Edge {
+    if (!this.#state) {
+      this.log('Invalid State - unable get Edges for Transistion');
```
Comment (robguy21) [inline src/machine.js]: log what `this.#state` is - it's super useful for debugging

**Review by robguy21 — CHANGES_REQUESTED** [review]

**Review by robguy21 — CHANGES_REQUESTED** [review]
Also please include tests

**Review by robguy21 — APPROVED** [review]

**Pull request merged**